compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -1729,51 +1730,58 @@ fn receiver_is_valid<'tcx>(
         autoderef = autoderef.include_raw_pointers();
     }
 
+    // The first type is `receiver_ty`, which we know its not equal to `self_ty`; skip it.
+    autoderef.next();
```

Comment:
This autoderef may have been load-bearing, since we're doing >1 equate per method even though we tested the receiver was equal above.